From e4201d7eda4b0b3c0384afb3e790be1dbd2e4bfe Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Thu, 3 Mar 2016 07:29:54 -0500
Subject: [PATCH 5/5] wr-policy: allow passwd_t, useradd_t search journal

Fix avc denials:

  avc: denied { search } for pid=905 comm="useradd" \
  name="journal" dev="tmpfs" ino=8209 \
  scontext=root:sysadm_r:useradd_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:syslogd_var_run_t:s15:c0.c1023 \
  tclass=dir permissive=0

  avc: denied { search } for pid=911 comm="passwd" \
  name="journal" dev="tmpfs" ino=8209 \
  scontext=root:sysadm_r:passwd_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:syslogd_var_run_t:s15:c0.c1023 \
  tclass=dir permissive=0

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/admin/usermanage.te | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/policy/modules/admin/usermanage.te b/policy/modules/admin/usermanage.te
index 74be7ef..65a1186 100644
--- a/policy/modules/admin/usermanage.te
+++ b/policy/modules/admin/usermanage.te
@@ -356,6 +356,8 @@ userdom_read_user_tmp_files(passwd_t)
 # on user home dir
 userdom_dontaudit_search_user_home_content(passwd_t)
 
+mls_file_read_all_levels(passwd_t)
+
 optional_policy(`
 	nscd_run(passwd_t, passwd_roles)
 ')
@@ -528,6 +530,8 @@ userdom_manage_user_home_content_files(useradd_t)
 userdom_home_filetrans_user_home_dir(useradd_t)
 userdom_user_home_dir_filetrans_user_home_content(useradd_t, notdevfile_class_set)
 
+mls_file_read_all_levels(useradd_t)
+
 optional_policy(`
 	mta_manage_spool(useradd_t)
 ')
-- 
1.9.1

